daemon/monitor.go

```diff
@@ -54,6 +54,9 @@ func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
 			defer autoRemove()
 		}
 
+		stateCtr.dec("running")
```

Comment:
Don't we need to wrap such `dec-then-inc` with mutex?
If `docker info` is called during dec and inc, it would result in weird output.